src/components/Drawer/Drawer.test.tsx

```diff
@@ -0,0 +1,115 @@
+import { screen } from '@testing-library/react';
+import React from 'react';
+
+import { renderWithProviders } from '../../utils/tests/renderWithProviders';
+import Drawer from './Drawer';
+import { Button } from '../Button';
+import { Icon } from '../Icon';
+import { Paragraph } from '../typography';
+import { Inline, Stack } from '../layout';
+import { pxToRem } from '../../utils/helpers';
+
+const Footer = () => (
+  <Inline>
+    <Button data-testid="footer">Submit</Button>
+  </Inline>
+);
+const Adornment = () => (
+  <div data-testid="adornment">
+    <Icon name="cog" color="info.500" />
+  </div>
+);
+
+describe('Drawer', () => {
+  it('should show footer if footer is defined', () => {
+    renderWithProviders(
+      <Drawer
+        size="sm"
+        onClose={() => null}
+        title="Test drawer"
+        footer={<Footer />}
+        dataTestId="drawer"
+      >
+        <Stack>
+          <Paragraph>Whatever happens, happens here</Paragraph>
+        </Stack>
+      </Drawer>,
+    );
+
+    const footer = screen.getByTestId('footer');
```

Comment:
You can actually test not the footer but the button it contains like:
```js
const footerButton = screen.getByRole('button',  { name: /Submit/i };
expect(footerButton).toBeInDocument();
```
https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#using-the-wrong-query